src/main/java/seedu/tr4cker/model/module/UniqueModuleList.java

```diff
@@ -0,0 +1,126 @@
+package seedu.tr4cker.model.module;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.tr4cker.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.tr4cker.model.module.exceptions.DuplicateModuleException;
+import seedu.tr4cker.model.module.exceptions.ModuleNotFoundException;
+
+
+/**
+ * A list of modules that enforces uniqueness between its elements and does not allow nulls.
+ * A module is considered unique by comparing using {@code Module#isSameModule(Module)}. As such, adding and
+ * updating of modules uses Module#isSameModule(Module) for equality so as to ensure that the module being added
+ * or updated is unique in terms of identity in the UniqueModuleList. However, the removal of a task uses
+ * n#equals(Object) so as to ensure that the module with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Module#isSameModule(Module)
+ */
+public class UniqueModuleList implements Iterable<Module> {
+
+    private final ObservableList<Module> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Module> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent Module as the given argument.
+     */
+    public boolean contains(Module toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameModule);
+    }
+
+    /**
+     * Returns true if the list contains a Module with the given module code.
+     */
+    public boolean containsCode(ModuleCode toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(x -> x.moduleCode.equals(toCheck));
+    }
+
+    /**
+     * Adds a module to the list.
+     * The module must not already exist in the list.
+     */
+    public void add(Module toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateModuleException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Removes the equivalent module from the list.
+     * The module must exist in the list.
+     */
+    public void remove(Module toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new ModuleNotFoundException();
+        }
+    }
+
+    public void setModules(UniqueModuleList replacement) {
```

Comment:
Missing JavaDocs for public method.

Reply:
![image](https://user-images.githubusercontent.com/69660935/97098947-c4b8af80-16bd-11eb-9f48-88181335bf4d.png)

Idk why it doesn't show its been updated

Reply:
oh can already!